src/shapes/itext.class.js

```diff
@@ -497,7 +497,7 @@
       ctx.fillRect(
         boundaries.left + leftOffset,
         boundaries.top + boundaries.topOffset,
-        this.cursorWidth / this.scaleX,
+        this.cursorWidth / this.scaleX / this.canvas.viewportTransform[0],
```

Comment:
Before fillRect, calculate a scale factor that take in account scaleX and this.canvas.getZoom();
var scaleFactor = this.scaleX;
if (this.canvas) {
  scaleFactor *= this.canvas.getZoom();
}

and then use that scaleFactor to divide the cursorWidth.